### DISCUSSION TITLE
Can't map keys from bluetooth keyboard in pocketbook

### DISCUSSION THREAD

**Discussion by Thegoban** [koreader/koreader#8703] [Q&A]
Pocketbook touch hd 3 can be paired with bluetooth buttons, I have paired a one hand gamepad for iPhone, it's correctly sending keystrokes as the stock reader is turning pages and changing chapters with no issues, but koreader is not. I've read a wiki  where It's explained how to map devices keys to change the order of the buttons creating  /koreader/settings/event_map.lua file, I wonder if that way it is supposed to work, as I need that feature.

I've been playing with it without success... maybe did't understand... don't know
is this the correct way (if it's possible), this is what I have inside my event_map.lua file

     Return { [272] = "RPgBack", [1] = "RPgFwd" }

Is this what it's supposed to be, provided that the numbers are correct?, should it work this or other way?

https://github.com/koreader/koreader/wiki/Keymapping

**Comment by Thegoban** [comment]
Well, after all, it seems that koreader is aware of the keystrokes, here is part of the the crash.log output

01/22/22-15:15:07 DEBUG AutoTurn: onInputEvent
01/22/22-15:15:07 DEBUG input event => type: 42 (nil), code: 272, value: 0, time: 1642860907.940239
01/22/22-15:15:07 DEBUG AutoStandby:onInputevent() instance= table: 0xa0263548
                                      .
                                      .
01/22/22-15:15:08 DEBUG AutoTurn: onInputEvent
01/22/22-15:15:09 DEBUG input event => type: 40 (nil), code: 1, value: 0, time: 1642860909.112940
01/22/22-15:15:09 DEBUG AutoStandby:onInputevent() instance= table: 0xa0263548
                                    
It sees codes 272 and 1, but it does nothing with them.

**Reply by NiLuJe** [reply]
That just means that now you know what to put in your custom key map ;).

**Reply by pazos** [reply]
Cross reference from https://www.mobileread.com/forums/showthread.php?t=344617

**Reply by Thegoban** [reply]
> That just means that now you know what to put in your custom key map ;).

show me a starting point to learn to do such thing , I'm open to investigation and learning

**Reply by pazos** [reply]
All is contained in https://www.mobileread.com/forums/showpost.php?p=4191985&postcount=6

Get a usbnet session. Tweak the file until it works. Send a PR.

**Reply by Thegoban** [reply]
It may be true that coders think that 1 Km are 1024 m.

**Reply by pazos** [reply]
Can't really eli5.

The best you can do is document what KOReader reports now.

Try different bluetooth devices and document each key:

For example:

#### volume up - headphones
`xx/xx/xx-xx:xx:xx DEBUG input event => type: 42 (nil), code: 272, value: 0, time: xxxxxxxxxx.xxxxxx`

#### button b - controller
`xx/xx/xx-xx:xx:xx DEBUG input event => type:  63 (nil), code: 666, value: 0, time: xxxxxxxxxx.xxxxxx`


Then wait for a future contributor that follows instructions in the mobileread link :)

That's basically:  The **type** of the event should be  `C.EV.KEY`
